web/src/stream_topic_history.ts

```diff
@@ -103,26 +123,26 @@ export class PerStreamHistory {
           deleted.
     */
 
-    topics = new FoldDict();
+    topics = new FoldDict<Topic>();
     // Most recent message ID for the stream.
     max_message_id = 0;
+    stream_id: number;
 
-    constructor(stream_id) {
+    constructor(stream_id: number) {
         this.stream_id = stream_id;
     }
 
-    has_topics() {
+    has_topics(): boolean {
         return this.topics.size !== 0;
     }
 
-    update_stream_max_message_id(message_id) {
+    update_stream_max_message_id(message_id: number): void {
         if (message_id > this.max_message_id) {
             this.max_message_id = message_id;
         }
     }
 
-    add_or_update({topic_name, message_id = 0}) {
-        message_id = Number.parseInt(message_id, 10);
+    add_or_update({topic_name, message_id}: {topic_name: string; message_id: number}): void {
```

Comment:
[here](https://github.com/zulip/zulip/pull/28414/commits/487e58b8bb09f13339aad7520e258d574e59091c#r1440882198) Tim said:

> Should this just take normal parameters like `maybe_remove` does? It only has one caller.

Reply:
Yeah that seems reasonable. I can make a prep commit for that